### PR TITLE
update info/variation/population REST endpoint to retrieve individuals

### PR DESCRIPTION
**Pull request by ima23**
### Description

The variation/population REST endpoint retrieves population names, descriptions and sample counts. Individuals/Population are also of interest. This update allows the retrieval of individual names and any genders for a given population name. 

### Use case

We have had requests for the ability to retrieve lists of individuals by population to help in analysis of genotype data. This functionality was only available in the perl API.

GET info/variation/populations/:species/:population_name

### Benefits

The update allows retrieval of all individuals for a population from a species.

### Possible Drawbacks

None.

### Testing

Yes. The corresponding tests have been updated.

_If so, do the tests pass/fail?_

Tests pass.

_Have you run the entire test suite and no regression was detected?_

Yes.

### Changelog

[/info/variation/population] Added the ability to retrieve all individuals for a population from a species.

### CODE REVIEW

**Comment by ima23** [comment]
This PR goes together with PR  https://github.com/Ensembl/ensembl-rest_private/pull/30.

**Comment by ima23** [comment]
Thanks @avullo for checking! Indeed the filter is not needed if data on a specific population is requested. I've cleaned the code in the next commit to reflect that and cleaned the parameter logic.

**Comment by avullo** [comment]
@ima23 thanks, this is good for me.

File: t/info.t [hunk 1]
```diff
@@ -274,6 +274,24 @@ is_json_GET(
...
+  my $population_name = "1000GENOMES:phase_1_ASW";
+  my $json = json_GET("/info/variation/populations/homo_sapiens/1000GENOMES:phase_1_ASW", 'GET specific population for given species in variation database');
+  cmp_ok(scalar(@{$json}), '==', 1, 'Test that one population was returned');
+  cmp_ok($json->[0]->{name}, 'eq', $population_name, 'Test that correct population was returned');
+  cmp_ok($json->[0]->{size}, '==', 61, 'Test correct population size');
+  cmp_ok(scalar(@{$json->[0]->{individuals}}), '==', 61, 'Test that correct number of individuals was returned');
+  cmp_ok($json->[0]->{individuals}->[0]->{name}, 'eq', '1000GENOMES:phase_1:NA19625', 'Test first individual name');
+  cmp_ok($json->[0]->{individuals}->[0]->{gender}, 'eq', 'Female', 'Test first individual gender');
+}
+
```
Comment (nerdstrike) [inline t/info.t]: I would like the species argument to be tested for regressions. e.g. species with no population data for the given group, and invalid species.

[the remark this answers is not in the document]
Comment (ima23) [reply t/info.t]: Thanks @nerdstrike for checking! All good questions. The new commit checks for request on existing species with no population data and invalid species name.

**Comment by avullo** [comment]
OK, CI complains with the usual rate limit stuff.

**Pull request merged**